build: expand the swift components and create install rules

compnerd opened the pull request:
Replace the component system with the LLVM installation mechanism.  This allows
for a single unified toolchain build.  Preserve the components by remapping the
swift components to their substiutients during the configure phase.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

gottesmm commented on CMakeLists.txt:
```diff
@@ -454,12 +454,14 @@ include(CMakePushCheckState)
 
 include(SwiftComponents)
+
+swift_expand_components()
```
Can you hide this in configure_components?

gottesmm reviewed (commented):
I want to review this. Also can you please not do inline changes for this. Why not just hide these changes in the underlying component system.

compnerd commented:
It just seems that having this be mapped away is easier to deal with the CMake logic overall.  Really, the component mapping doesn't change very often, and most of that can be abstracted over with caches, so this seems better for the regular read/modify cycle of the CMake.

futurejones commented:
Can one of the admins verify this patch?

shahmishal commented:
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

compnerd committed:
utils: extend the experimental runtime build

This adjusts the experimental runtime build to include the overlay,
StringProcessing, Synchronization. This is a first step towards getting
a sufficient amount of the SDK overlay built so that we can start
statically linking a real world program (swift-driver).
